tagstudio/src/core/library.py

```diff
@@ -86,20 +86,20 @@ def __repr__(self) -> str:
         return self.__str__()
 
     def __eq__(self, __value: object) -> bool:
```

Comment:
At this point there's no type check in this method. I'm not sure how how is that better than casting the variable, but if you want disable the type checking then you can add this decorator to the method declaration rather than adding `ignore` on each line with error.

```suggestion
    @typing.no_type_check
    def __eq__(self, __value: object) -> bool:
```

Reply:
Thanks for the info on the `@typing.no_type_check` decorator, I'll definitely use that if needed going forward 👍 

The casting itself was causing me the following error when invoking the \_\_eq\_\_ method:
```py
Traceback (most recent call last):
  File "F:\GitHub\TagStudio/tagstudio\src\qt\helpers\custom_runnable.py", line 18, in run
    self.function()
  File "F:\GitHub\TagStudio/tagstudio\src\qt\modals\merge_dupe_entries.py", line 44, in <lambda>
    r = CustomRunnable(lambda: iterator.run())
                               ^^^^^^^^^^^^^^
  File "F:\GitHub\TagStudio/tagstudio\src\qt\helpers\function_iterator.py", line 20, in run
    for i in self.iterable():
  File "F:\GitHub\TagStudio/tagstudio\src\core\library.py", line 1043, in merge_dupe_entries
    self.entries.remove(id_to_entry_map[id])
  File "F:\GitHub\TagStudio/tagstudio\src\core\library.py", line 92, in __eq__
    int(self.id) == int(__value.id)
                        ^^^^^^^^^^
AttributeError: type object 'object' has no attribute 'id'
```
I can reproduce the error by comparing any two Entry objects:
```py
entry_1 = Entry(1, "file_2", "path_2", [])
        entry_2 = Entry(2, "file_2", "path_2", [])
        logging.info(entry_1 == entry_2)
```

It looks like the object cast is just stripping the Entry's attributes down and turning it back into a bare object, right? I'm not sure how the cast is intended to function here.

Reply:
right, my bad. It should be this instead:
```
__value = cast(Self, __value)
```